packages/tools/src/tools/VolumeCroppingControlTool.ts

```diff
@@ -0,0 +1,1771 @@
+      this.setToolCenter(newMax, 'max');
+    }
+  };
+
+  _onNewVolume = () => {
+    const viewportsInfo = this._getViewportsInfo();
+    if (viewportsInfo && viewportsInfo.length > 0) {
+      const { viewportId, renderingEngineId } = viewportsInfo[0];
+      const renderingEngine = getRenderingEngine(renderingEngineId);
+      const viewport = renderingEngine.getViewport(viewportId);
+      const volumeActors = viewport.getActors();
+      if (volumeActors.length > 0) {
+        const imageData = volumeActors[0].actor.getMapper().getInputData();
+        if (imageData) {
+          const dimensions = imageData.getDimensions();
+          const spacing = imageData.getSpacing();
+          const origin = imageData.getOrigin();
+          const cropFactor = this.configuration.initialCropFactor ?? 0.2;
+
+          this.toolCenter = [
+            origin[0] + cropFactor * (dimensions[0] - 1) * spacing[0],
+            origin[1] + cropFactor * (dimensions[1] - 1) * spacing[1],
+            origin[2] + cropFactor * (dimensions[2] - 1) * spacing[2],
+          ];
+          const maxCropFactor = 1 - cropFactor;
+          this.toolCenterMin = [
+            origin[0] + cropFactor * (dimensions[0] - 1) * spacing[0],
+            origin[1] + cropFactor * (dimensions[1] - 1) * spacing[1],
+            origin[2] + cropFactor * (dimensions[2] - 1) * spacing[2],
+          ];
+          this.toolCenterMax = [
+            origin[0] + maxCropFactor * (dimensions[0] - 1) * spacing[0],
+            origin[1] + maxCropFactor * (dimensions[1] - 1) * spacing[1],
+            origin[2] + maxCropFactor * (dimensions[2] - 1) * spacing[2],
+          ];
+          // Update all annotations' handles.toolCenter
+          const annotations =
+            getAnnotations(this.getToolName(), viewportId) || [];
+          annotations.forEach((annotation) => {
+            if (annotation.data && annotation.data.handles) {
+              annotation.data.handles.toolCenter = [...this.toolCenter];
+            }
+          });
+        }
+      }
+    }
+    this._computeToolCenter(viewportsInfo);
+  };
+
+  _unsubscribeToViewportNewVolumeSet(viewportsInfo) {
+    viewportsInfo.forEach(({ viewportId, renderingEngineId }) => {
+      const { viewport } = getEnabledElementByIds(
+        viewportId,
+        renderingEngineId
+      );
+      const { element } = viewport;
+
+      element.removeEventListener(
+        Enums.Events.VOLUME_VIEWPORT_NEW_VOLUME,
+        this._onNewVolume
+      );
+    });
+  }
+
+  _subscribeToViewportNewVolumeSet(viewports) {
+    viewports.forEach(({ viewportId, renderingEngineId }) => {
+      const { viewport } = getEnabledElementByIds(
+        viewportId,
+        renderingEngineId
+      );
+      const { element } = viewport;
+
+      element.addEventListener(
+        Enums.Events.VOLUME_VIEWPORT_NEW_VOLUME,
+        this._onNewVolume
+      );
+    });
+  }
+
+  _areViewportIdArraysEqual = (viewportIdArrayOne, viewportIdArrayTwo) => {
```

Comment:
The function always returns true unless the lengths differ or the last item is not found.
If any item is not found, it returns false from inside the forEach, but this does not break out of the function, so the function will still return true at the end. Maybe you can use a for loop or every instead of forEach for early return.